source/geod24/LocalRest.d

```diff
@@ -823,6 +847,46 @@ unittest
     test.ctrl.shutdown();
 }
 
+/// Example where a shutdown() routine must be called on a node before
+/// its destructor is called
+unittest
+{
+    __gshared bool dtor_called;
+    __gshared bool shutdown_called;
+    __gshared bool onDestroy_called;
+
+    static interface API
+    {
+        @safe:
+        public @property ulong pubkey ();
+    }
+
+    static class MockAPI : API
+    {
+        @safe public override @property ulong pubkey ()
+        { return 42; }
+        public void shutdown () { shutdown_called = true; }
+        ~this () { dtor_called = true; }
+    }
+
+    static void onDestroy (API node)
+    {
+        assert(!dtor_called);
+        auto mock = cast(MockAPI)node;
+        assert(mock !is null);
+        mock.shutdown();
+        onDestroy_called = true;
+    }
+
+    scope test = RemoteAPI!API.spawn!MockAPI();
+    assert(test.pubkey() == 42);
+    test.ctrl.shutdown(&onDestroy);
+    core.thread.Thread.sleep(1.seconds);  // ctr.shutdown call is asynchronous
```

Comment:
```suggestion
    while (!core.atomic.atomicLoad(dtor_called)) core.atomic.pause();
```

The problem is that this test doesn't fail, but at least it won't spuriously fail.

Reply:
> core.atomic.pause

hello there

Cool. Didn't know about this.

Reply:
We should fix other places where we have sleep calls like this.

Reply:
Is `core.atomic.pause` a new feature? I'm getting a build failure on ldc 14 and DMD 2.085.1

Reply:
Oh those are quite old compilers too. But I assume you still want to support them.

Reply:
Yeah a `sleep` in a loop might just be simpler

Reply:
My main concern was the lack of a retry mechanism